eZ/Publish/Core/Persistence/Cache/NotificationHandler.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\Persistence\Cache;
+
+use eZ\Publish\SPI\Persistence\Notification\Handler;
+use eZ\Publish\SPI\Persistence\Notification\Notification;
+
+class NotificationHandler extends AbstractHandler implements Handler
```

Comment:
After rethinking this will methods will be heavily used, so the proper caching is necessary (especially `countPendingNotificationsByOwnerId`).